packages/node/src/app/settings.ts

```diff
@@ -40,6 +41,10 @@ export interface AnalyticsSettings {
    * Default: an HTTP client that uses globalThis.fetch, with node-fetch as a fallback.
    */
   httpClient?: HTTPFetchFn | HTTPClient
+  /**
+   * Set up OAuth2 authentication between the client and Segment's endpoints
+   */
+  oauthSettings?: OAuthSettings
```

Comment:
I'm not sure it makes sense to include all the settings `TokenManager` uses - specifically the HTTPClient since we already let the user specify the httpClient at the top level.

What are your thoughts on tightening this a bit?

```
type OAuthSettings = Pick<TokenManagerSettings, 'clientId' | 'clientKey' | 'keyId' | 'authServer' | 'scope' | 'maxRetries'>
  oauthSettings?: OAuthSettings
```

Reply:
@chrisradek one tangent: I'd like to make OAuthSettings a neat interface with JSDoc and all that, so it looks as nice as:
https://github.com/segmentio/analytics-next/blob/turborepo-helper-input/packages/node/src/app/settings.ts. Using "Pick" would make things DRY, but would also make it harder to do that.

Reply:
I agree with this as a general pattern we should follow.

Reply:
@MichaelGHSeg made a case to support the HTTPClient, so I will withdraw my comment.